### PR TITLE
Marker selection table

### PR DESCRIPTION
**Pull request by cataphract**

### CODE REVIEW

**Comment by cataphract** [comment]
Companion pull request: https://github.com/thehyve/heim-transmartApp/pull/2

File: web-app/js/smartR/heatmapService.js [hunk 1]
```diff
@@ -257,16 +261,45 @@ window.HeatmapService = (function(smartRHeatmap){
         console.log('service.runAnalysis', params);
         var defer = jQuery.Deferred();
 
-        function showD3HeatMap(data) {
-            downloadJsonFile(this.executionId, 'heatmap.json')
-                    .then(function(d) { smartRHeatmap.create(d); })
-                    .then(function() { defer.resolve(); });
+        function runAnalysisSuccess(data) {
+            var i = 2;
```
Comment (rnugraha) [inline web-app/js/smartR/heatmapService.js]: what number is this?

**Comment by cataphract** [comment]
Dummy data: https://gist.github.com/256ff8f45daaf64e3fca

`run.R` should either produce this file (and name if `markerSelectionTable.json`) or **delete** it.

File: web-app/js/smartR/heatmapService.js (hunk 1 above)
Comment (cataphract) [inline web-app/js/smartR/heatmapService.js]: We need two pieces of data -- the heatmap and the table. We only resolve the deferred after we get both. So it's sort of like using `$.when()`

**Pull request merged**